src/span_buffer.cpp

```diff
@@ -37,12 +41,15 @@ void WritingSpanBuffer::registerSpan(const SpanContext& context) {
   std::lock_guard<std::mutex> lock_guard{mutex_};
   uint64_t trace_id = context.traceId();
   auto trace = traces_.find(trace_id);
-  if (trace == traces_.end()) {
+  if (trace == traces_.end() || trace->second.all_spans.empty()) {
```

Comment:
What is this extra condition for? I'd think it never happens, because the all_spans only gets created when a PendingTrace is created in response to registering a new span, so there should always be one!

Reply:
Tests.

```
  auto buffer = std::make_shared<MockBuffer>();
  buffer->traces()[123].sampling_priority = ....;
```
When the test does this kind of thing, the trace exists but without any spans.
So it doesn't apply the "firs span for a trace" behaviour.